utils/clients/ratios/client.go

```diff
@@ -65,6 +88,12 @@ type RateResponse struct {
 
 // FetchRate fetches the rate of a currency to BAT
 func (c *HTTPClient) FetchRate(ctx context.Context, base string, currency string) (*RateResponse, error) {
+	var cacheKey = fmt.Sprintf("%s_%s", base, currency)
+	// check cache for this rate
+	if rate, found := c.cache.Get(cacheKey); found {
```

Comment:
given issues we've had in the past, it may be a good idea to add a stale if error behavior

Reply:
+1